### PR TITLE
Add a `get_partial_path` method to NodePath

### LINKED ISSUE #7148: Add a `get_partial_path` method to NodePath
### Describe the project you are working on

A game with a lot of dynamic code.

### Describe the problem or limitation you are having in your project

Currently, if you have a NodePath like `^"A/B/C/D/E/F"`, there is no easy way to get only the first X names in the path, such as if you only wanted the first 4 names `^"A/B/C/D"`.

The only way to get this would be to manually reconstruct it in GDScript. However, this would be extremely inefficient compared to what's optimal, since the NodePath constructor that accepts an array of StringNames is not exposed, it's only internal to the engine, so you would have to build a String which then gets split up again. Horrible!

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Add a method that allows getting a partial path from a NodePath.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

See this PR: https://github.com/godotengine/godot/pull/78638

### If this enhancement will not be used often, can it be worked around with a few lines of script?

It can be worked around but not in a few lines and extremely inefficiently.

I don't know how often this will be used. I suspect that this will not be used often, but I have no idea.

### Is there a reason why this should be core and not an add-on in the asset library?

Yes, because this is heavily related to NodePath and NodePath is a part of core. The ability to manipulate NodePaths to get other NodePaths seems like a very core feature that NodePath should have.

### PR DESCRIPTION
**Pull request by aaronfranke**
Implements this proposal and closes https://github.com/godotengine/godot-proposals/issues/7148

This PR adds a `get_partial_path` method to NodePath for getting the first X parts of a path. Discussion: Would it be worth specifying the start index too, or getting the last parts of a path? I only need the first parts for my uses.

I added some test cases to ensure the behavior is correct, and I ran into a minor bug where the internal constructors with the StringName arrays will discard the absolute flag if the path is empty. So I added a check to continue with the construction whenever absolute is set to true (return only when absolute is set to false).

### CODE REVIEW

File: core/string/node_path.cpp [hunk 1]
```diff
@@ -331,7 +346,7 @@ NodePath NodePath::simplified() const {
 }
 
 NodePath::NodePath(const Vector<StringName> &p_path, bool p_absolute) {
-	if (p_path.size() == 0) {
+	if (p_path.size() == 0 && !p_absolute) {
```
Comment (nlupugla) [inline core/string/node_path.cpp:349]: What is the reason for this change and the similar one below? Do I need to include a similar change when making the slice method?

Comment (aaronfranke) [reply core/string/node_path.cpp:349]: The problem was an edge case where `p_absolute` was being ignored if the path was empty.

**Comment by nlupugla** [comment]
@aaronfranke Finished writing up my pr :) https://github.com/godotengine/godot/pull/81822

**Comment by aaronfranke** [comment]
Superseded by https://github.com/godotengine/godot/pull/81822
